library/core/tests/num/mod.rs

```diff
@@ -126,15 +126,15 @@ fn test_can_not_overflow() {
     where
         T: std::convert::TryFrom<i8>,
     {
-        !can_not_overflow::<T>(radix, T::try_from(-1_i8).is_ok(), input.as_bytes())
+        safe_width::<T>(radix, T::try_from(-1_i8).is_ok()) < input.len()
     }
 
     // Positive tests:
```

Comment:
suggestion: how about testing the output of safe_width directly?  Just seeing `can_overflow` returning `true` doesn't mean that it's correct -- it could be returning `usize::MAX`.